### PR TITLE
🩹🚨 - fix for redirect with query params

### PR DESCRIPTION
**Pull request by gilwo**
## Redirect middleware not redirecting url with query params

fix redirect to add query params (if applicable) to the redirected url, add tests.

## Type of change

Please delete options that are not relevant.

- [x] Bug fix (non-breaking change which fixes an issue)

## Checklist:

- [x] I have performed a self-review of my own code
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes

## Commit formatting:

Use emojis on commit messages so it provides an easy way of identifying the purpose or intention of a commit. Check out the emoji cheatsheet here: https://gitmoji.carloscuesta.me/

### CODE REVIEW

**Comment by gilwo** [comment]
@nickajacks1 - I looked into `c.RedirectToRoute()` - i find it cumbersome for the purpose of redirect with query params.
it might be useful in case the redirect require additional  (or modified) query params, but this is a new feature if i am not mistaken. 
it require a bit more changes in the redirect middleware (especially in the config). my purpose was to fix a flow which appear mandatory in my opinion - the extra complication was not required in my use case.

**Comment by nickajacks1** [comment]
>  i find it cumbersome for the purpose of redirect with query params.
You should be able to just set the params to `fiber.Map{"queries": c.Queries()}` right?
Well, you can't actually do that right now because `Redirect` doesn't accept those.

Either way, i wonder what the consequence of redirecting query params is..if the same query parameters have different meanings in the original and redirect site, you might get unexpected results (in which case the solution would probably be to do the redirect manually instead of using the middleware)
`www.example.com/files.zip?explode=true` => `www.example.com/bomb?explode=true`

This change seems ok, need more opinions though.

File: middleware/redirect/redirect_test.go [hunk 1]
```diff
@@ -104,6 +110,12 @@ func Test_Redirect(t *testing.T) {
 			url:        "/api/test",
 			statusCode: fiber.StatusOK,
 		},
+		{
+			name:       "redirect with query params",
+			url:        "/params?quary=abc",
```
Comment (nickajacks1) [inline middleware/redirect/redirect_test.go:115]: typo?
quary => query

File: middleware/redirect/redirect_test.go [hunk 2]
```diff
@@ -104,6 +110,12 @@ func Test_Redirect(t *testing.T) {
 			url:        "/api/test",
 			statusCode: fiber.StatusOK,
 		},
+		{
+			name:       "redirect with query params",
+			url:        "/params?query=abc",
+			redirectTo: "/with_params?query=abc",
+			statusCode: fiber.StatusMovedPermanently,
+		},
```
[the remark this answers is not in the document]
Comment (gilwo) [reply middleware/redirect/redirect_test.go:118]: i see no reason for multiple args independent check  as `QueryString()` already handle multiple query args (see [here](https://github.com/valyala/fasthttp/blob/12949de784e688451205333ebaf87ec0ca16927d/args_test.go#L285))

**Review by ReneWerner87 — APPROVED** [review]

**Review by ReneWerner87 — APPROVED** [review]

**Pull request merged**